modules/practice_adapter.js

```diff
@@ -300,6 +300,13 @@ class PracticeAdapter extends EventEmitter {
 
   async postLeaderboard (leaderboard) {
     if (this._config.get('postLeaderboardOnReset')) {
+      // First unpin any old leaderboards
+      // Assume that all pinned messages in the announcementChannel by the bot are leaderboards
+      this._announcementsChannel.messages.fetchPinned()
```

Comment:
You're gonna want to await this, otherwise the rest of the function may complete (and pin the new LB) and then the new LB immediately gets unpinned.

Reply:
Is that not the purpose of the then chain? Seems to have behaved fine.

Reply:
Oh, actually, it's fine, but for slightly different reasons.

The `then()` callback can run at any time, but it will use the pins that were retrieved at the time fetchPinned was called, which doesn't include the new LB.